### DISCUSSION TITLE
fatal: Could not read from remote repository.

### DISCUSSION THREAD

**Discussion by claytoncalixto** [community/community#114595] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

Help friends!
I can´t to do sent first commit with git push for github repository, see:

Clayton Calixto@DESKTOP-OEPNC3H MINGW64 /d/SSD/eclipse-projetos/cadastro-de-pessoas-sinerji (main)
$ git push --set-upstream origin main
ssh: connect to host gihub.com port 22: Connection timed out
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.

I alredy change key ssh, to do autentication two factor, but nothing.

![image](https://github.com/community/community/assets/27165014/7dc276db-555a-4a9c-ac85-a66ea00b3846)

Can you help I, please?

**Comment by nehalp13** [comment]
Try this following steps :
1) Verify SSH Configuration: Double-check your SSH configuration to make sure that you've correctly set up your SSH key with GitHub. You can verify this by running the following command:
ssh -T git@github.com
If successful, you should see a message like "Hi username! You've successfully authenticated...".
2) Firewall and Proxy Settings: If you're behind a firewall or using a proxy server, make sure that it's configured to allow SSH connections to GitHub's servers on port 22. Sometimes, corporate or institutional networks may have restrictions that prevent outbound SSH connections.
3) Try HTTPS: As an alternative, you can try pushing your commits using HTTPS instead of SSH. GitHub supports both SSH and HTTPS for Git operations. You can switch the remote URL to use HTTPS by running the following command:
git remote set-url origin https://github.com/username/repository.git
Replace "username/repository" with your actual GitHub username and repository name.
